addons/binding/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/handler/IhcHandler.java

```diff
@@ -413,35 +419,41 @@ private void sendPulseCommand(ChannelUID channelUID, ChannelParams params, WSRes
         logger.debug("Channel params: {}", params);
         Converter<WSResourceValue, Type> converter = ConverterFactory.getInstance().getConverter(value.getClass(),
                 OnOffType.class);
-        ConverterAdditionalInfo converterAdditionalInfo = new ConverterAdditionalInfo(null, params.isInverted());
 
-        WSResourceValue valOn = converter.convertFromOHType(OnOffType.ON, value, converterAdditionalInfo);
-        WSResourceValue valOff = converter.convertFromOHType(OnOffType.OFF, value, converterAdditionalInfo);
+        if (converter != null) {
+            ConverterAdditionalInfo converterAdditionalInfo = new ConverterAdditionalInfo(null, params.isInverted(),
+                    getCommandLevels(params));
 
-        // set resource to ON
-        logger.debug("Update resource value (inverted output={}): {}", params.isInverted(), valOn);
-        if (updateResource(valOn)) {
-            scheduler.submit(() -> {
-                // sleep a while
-                try {
-                    logger.debug("Sleeping: {}ms", pulseWidth);
-                    Thread.sleep(pulseWidth);
-                } catch (InterruptedException e) {
-                    // do nothing
-                }
-                // set resource back to OFF
+            WSResourceValue valOn = converter.convertFromOHType(OnOffType.ON, value, converterAdditionalInfo);
+            WSResourceValue valOff = converter.convertFromOHType(OnOffType.OFF, value, converterAdditionalInfo);
 
-                logger.debug("Update resource value (inverted output={}): {}", params.isInverted(), valOff);
-                try {
-                    if (!updateResource(valOff)) {
-                        logger.warn("Channel {} update to resource '{}' failed.", channelUID, valOff);
+            // set resource to ON
+            logger.debug("Update resource value (inverted output={}): {}", params.isInverted(), valOn);
+            if (updateResource(valOn)) {
+                scheduler.submit(() -> {
+                    // sleep a while
+                    try {
+                        logger.debug("Sleeping: {}ms", pulseWidth);
+                        Thread.sleep(pulseWidth);
```

Comment:
This was also already in the code. But was there any reason not to just use `scheduler.schedule` instead of submit with sleep? sleep is considered bad because this schedule is on a limited thread pool also used by other bindings. So this sleep blocks a thread possible reducing the number of threads available to other bindings, possible degrading performance.

Reply:
I will do another fix for that later on, when I have more time to test it.